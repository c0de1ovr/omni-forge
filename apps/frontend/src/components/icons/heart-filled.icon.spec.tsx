import { render } from '@testing-library/react';

import { HeartFilledIcon } from './heart-filled.icon';

describe('heart filled icon', () => {
  it('should render heart icon correctly', () => {
    const { container } = render(<HeartFilledIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should accept and use height', () => {
    const { container } = render(<HeartFilledIcon width={10} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should accept and use width', () => {
    const { container } = render(<HeartFilledIcon height={10} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
