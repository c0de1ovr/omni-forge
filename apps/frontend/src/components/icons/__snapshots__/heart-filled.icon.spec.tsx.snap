// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`heart filled icon should accept and use height 1`] = `
<svg
  aria-hidden="true"
  focusable="false"
  height="24"
  role="presentation"
  viewBox="0 0 24 24"
  width="10"
>
  <path
    d="M12.62 20.81c-.34.12-.9.12-1.24 0C8.48 19.82 2 15.69 2 8.69 2 5.6 4.49 3.1 7.56 3.1c1.82 0 3.43.88 4.44 2.24a5.53 5.53 0 0 1 4.44-2.24C19.51 3.1 22 5.6 22 8.69c0 7-6.48 11.13-9.38 12.12Z"
    fill="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
  />
</svg>
`;

exports[`heart filled icon should accept and use width 1`] = `
<svg
  aria-hidden="true"
  focusable="false"
  height="10"
  role="presentation"
  viewBox="0 0 24 24"
  width="24"
>
  <path
    d="M12.62 20.81c-.34.12-.9.12-1.24 0C8.48 19.82 2 15.69 2 8.69 2 5.6 4.49 3.1 7.56 3.1c1.82 0 3.43.88 4.44 2.24a5.53 5.53 0 0 1 4.44-2.24C19.51 3.1 22 5.6 22 8.69c0 7-6.48 11.13-9.38 12.12Z"
    fill="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
  />
</svg>
`;

exports[`heart filled icon should render heart icon correctly 1`] = `
<svg
  aria-hidden="true"
  focusable="false"
  height="24"
  role="presentation"
  viewBox="0 0 24 24"
  width="24"
>
  <path
    d="M12.62 20.81c-.34.12-.9.12-1.24 0C8.48 19.82 2 15.69 2 8.69 2 5.6 4.49 3.1 7.56 3.1c1.82 0 3.43.88 4.44 2.24a5.53 5.53 0 0 1 4.44-2.24C19.51 3.1 22 5.6 22 8.69c0 7-6.48 11.13-9.38 12.12Z"
    fill="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="1.5"
  />
</svg>
`;
